File: 백준/Gold/9935. 문자열 폭발/문자열 폭발.py
import sys

input = sys.stdin.readline
string = input().strip()
bomb = input().strip()

# 문자를 돌면서 폭발 문자열이 있는 지 찾는다.
# 폭발 문자열이 있으면 그 문자열을 제거하고 다시 폭발 문자열이 있는 지 찾는다.
# 반복!

stack = []
bomb_length = (len(bomb))

for i in range(len(string)):
    stack.append(string[i])
    
    if "".join(stack[-bomb_length:]) == bomb:
        for _ in range(bomb_length):
            stack.pop()
    
    
    
    
if stack:
    print("".join(stack))
else:
    print("FRULA")

File: 백준/Gold/9935. 문자열 폭발/문자열 폭발.cc
#include <iostream>
#include <stack>
#include <queue>

using namespace std;
// 스택의 폭발물의 길이만큼이 같은 지 확인하고 같으면 폭발!

int main() {
    string input, bomb;
    cin >> input >> bomb;

    vector<char> answer;

    for (char c : input) {
        answer.push_back(c);

        if (answer.size() >= bomb.size()) {
            bool match = true;

            for (int i = 0; i < bomb.size(); i++) {
                if (answer[answer.size() - bomb.size() + i] != bomb[i]) {
                    match = false;
                    break;
                }
            }

            if (match) {
                for (int i = 0; i < bomb.size(); i++) {
                    answer.pop_back();
                }
            }
        }
    }

    if (!answer.empty()) {
        for (char c : answer) {
            cout << c;
        }
        cout << endl;
    } else {
        cout << "FRULA" << endl;
    }

    return 0;
}